src/RequestParser.php

```diff
@@ -79,41 +94,48 @@ private function createModelBuilderStruct(Request $request): ModelBuilderStruct
 
     private function getBaseModelName(Request $request): string
     {
-        $modelCandidates = [];
-        $route = $request->route();
-        $controller = $this->getControllerFromRoute($route);
-        if ($controller) {
-            $stringToRemove = "controller";
-            $className = class_basename($controller);
-            $maybeBaseModel = substr_replace($className, '', strrpos(strtolower($className), $stringToRemove), strlen($stringToRemove));
-            $modelCandidates[] = $maybeBaseModel;
-
-            $modelName = $this->getModelFromNamespaces($maybeBaseModel, $this->modelNamespaces);
-            if ($modelName) {
-                return $modelName;
-            }
+        if ($this->modelName) {
+            return $this->modelName;
         }
 
-        $exploded = explode("/", $request->getRequesturi());
-        $lastURISegment = strtolower(end($exploded));
-        $lastURINoQuery = current(explode("?", $lastURISegment, 2));
-        $camelizeURI = str_replace('_', '', ucwords($lastURINoQuery, '_'));
-        $modelCandidates[] = $camelizeURI;
+        $errorMessage = 'Model not found, please use "setModel()" method.';
+        if ($this->guessModelName) {
```

Comment:
Where do we set `guessModelName` to false ?
If we never set that, let's remove this line.

Reply:
Oh, got it. So I believe this line also not needed any longer, right?
```
$errorMessage = 'Model not found, please use "setModel()" method.';
```

Reply:
Depend on the implementation.
If you want to set the `guessModelName` to false, I think you can do it on `setModel` method.